provider/github/watcher.go

```diff
@@ -73,7 +73,18 @@ func (w *Watcher) Watch(ctx context.Context, cb lookout.EventHandler) error {
 func (w *Watcher) watchPrs(ctx context.Context, c *Client, repos []*lookout.RepositoryInfo, cb lookout.EventHandler, errCh chan error) {
 	for {
 		for _, repo := range repos {
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(w.newInterval(c.Rate(coreCategory))):
```

Comment:
could you please double check it works correctly?
the problem is before we were calling `c.Rate` after a request is done, now before. So it returns empty value. As I remember I didn't really test it with empty value.

Reply:
Yes, with zero values it returns the `minInterval`